src/parser.js

```diff
@@ -2101,27 +2141,52 @@ export class GenericParser extends Tokenizer {
 
   parseObjectExpression() {
     let startState = this.startNode();
-
     this.lex();
-
+    let rest = null;
     let properties = [];
     while (!this.match(TokenType.RBRACE)) {
-      let property = this.inheritCoverGrammar(this.parsePropertyDefinition);
-      properties.push(property);
+      let property = null;
+      let isSpreadProperty = false;
+      let spreadPropertyOrAssignmentTarget = null;
```

Comment:
This should be inside of the `if` - `let spreadPropertyOrAssignmentTarget = this.parseSpreadPropertyDefinition();`